GDJS/Runtime/events-tools/networktools.ts

```diff
@@ -66,6 +70,43 @@ namespace gdjs {
         }
       };
 
+      export const sendAwaitableAsyncRequest = (
+        url: string,
+        body: string,
+        method: string,
+        contentType: string,
+        responseVar: gdjs.Variable,
+        errorVar: gdjs.Variable
+      ) => {
+        return new gdjs.PromiseTask(
+          fetch(url, {
```

Comment:
I used fetch here since it returns a promise by default so it is more convenient to make an asynchronous action with. While I tried to keep the behavior the same as with the XmlHttpRequest, there might be some differences in internal behavior still. Is it fine to keep it like this or should I change it to use XmlHttpRequest to be 100% certain that the behavior is the same as the synchronous version?

Reply:
I think fetch is fine! Its support is now more than enough even across dated browsers, and I don't think we risk much in terms of behavior.